reservation/reservation.go

```diff
@@ -106,6 +106,16 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 	return res, nil
 }
 
+// WaitUntilLock creates a Reservation for `resource`, or waits until it can do so.
+func (manager *Manager) WaitUntilLock(resource string) (*Reservation, error) {
+	res, err := manager.Lock(resource)
+	for err != nil && err.Error() == fmt.Sprintf("Reservation already exists for resource %s", resource) {
```

Comment:
I pattern that I've seen pretty commonly for these sorts of "do while" loops in Go looks like this:

```go
var res *Reservation
var err error
for res, err = manager.Lock(resource); err != nil; res, err = manager.Lock(resource) {
  time.Sleep(time.Second)
}
```